framework/sysdeps/unix/child_debug.cpp

```diff
@@ -210,14 +210,18 @@ static void child_crash_handler(int, siginfo_t *si, void *ucontext)
                 futex_wait(&thread_state, int(thread_failed));
         }
 
-        // restore the signal handler so we can be killed
-        signal(si->si_signo, SIG_DFL);
-        if (si->si_signo != SIGABRT) {
+        // restore the signal handler so the process can be killed
+        signal(sig, SIG_DFL);
+
+        // non-ABORT FROMKERNEL only
+        if ((sig != SIGABRT) && si && (si->si_code > 0)) {
```

Comment:
Do we need to check the signal number? Wouldn't it be enough to get a `>= SI_KERNEL` here? The kernel will not send SIGABRT.